### DISCUSSION TITLE
Server crash on startup.

### DISCUSSION THREAD

**Discussion by Timtendo12** [rabbitmq/rabbitmq-server#7655] [Other] [answered]
Hi, Im running RabbitMQ on docker through pterodactyl panel (server/docker management), However whenever I start the server it immediately crashes with an error that really does not explain much whats going wrong..

```
container@pterodactyl~ ./sbin/rabbitmq-server
{"init terminating in do_boot",{undef,[{rabbit,boot,[],[]},{init,start_em,1,[]},{init,do_boot,3,[]}]}}
init terminating in do_boot ({undef,[{rabbit,boot,[],[]},{init,start_em,1,[]},{init,do_boot,3,[]}]})

Crash dump is being written to: erl_crash.dump...done
```

- Using erlang 24,
- with the latest version of RabbitMQ.
- Docker is ran on port 5672

Installed using this Pterodactyl egg: https://github.com/parkervcp/eggs/tree/master/software/rabbitmq (an egg is basically an installation script for Pterodactyl)

**crash dump:**
[erl_crash.dump.txt](https://github.com/rabbitmq/rabbitmq-server/files/10992779/erl_crash.dump.txt)

**Comment by michaelklishin** [comment, marked as answer]
`rabbit:boot/0` cannot be "undefined" (`undef`), so this means that the runtime cannot load the module. 

Have a look at what Erlang version is required by the "latest RabbitMQ release" (all 3.11.x versions): it is [Erlang 25](https://www.rabbitmq.com/which-erlang.html).

**Reply by Timtendo12** [reply]
Stupid thing of me. It was indeed the outdated erlang version. Updating it to 26 worked.

**Reply by michaelklishin** [reply]
26 has not shipped yet and has at least one incompatibility with one of RabbitMQ's dependencies. Consider using Erlang 25.x.

**Comment by cihatsolak** [comment]


**Reply by Timtendo12** [reply]
Stupid thing of me. It was indeed the outdated erlang version. Updating it to 26 worked.

**Reply by lukebakken** [reply]
@cihatsolak your answer reads as though it came from ChatGPT.
